org.eclipse.xtext/src/org/eclipse/xtext/generator/trace/node/TracedAccessorsProcessor.java

```diff
@@ -49,14 +50,14 @@ public void doTransform(MutableClassDeclaration annotatedClass, @Extension Trans
 		for (InterfaceDeclaration f : Iterables.filter(ListExtensions.map(Arrays.asList(factories), it -> it.getType()),
 				InterfaceDeclaration.class)) {
 			for (TypeReference t : IterableExtensions.map(
-					IterableExtensions
+					IterableExtensions.sortBy(IterableExtensions
 							.filter(f.getDeclaredMethods(),
 									it -> it.getSimpleName().startsWith("create")
-											&& IterableExtensions.isEmpty(it.getParameters())),
+											&& IterableExtensions.isEmpty(it.getParameters())), MethodDeclaration::getSimpleName),
 					it -> it.getReturnType())) {
-				for (ResolvedMethod getter : IterableExtensions.filter(
+				for (ResolvedMethod getter : IterableExtensions.sortBy(IterableExtensions.filter(
 						IterableExtensions.filter(t.getAllResolvedMethods(), it -> isSupportedGetter(it)),
-						it -> !iterableType.isAssignableFrom(it.getDeclaration().getReturnType()))) {
+						it -> !iterableType.isAssignableFrom(it.getDeclaration().getReturnType())), ResolvedMethod::getSimpleSignature)) {
```

Comment:
It's a little sad that it's a different terminology here and above (name vs signature). But it is how it is.

I'd really prefer a small refactoring towards FluentIterable to have a better flow when reading the code.
Event though it does not save much space:

```
IterableExtensions.map(
					IterableExtensions
					IterableExtensions.sortBy(IterableExtensions
							.filter(f.getDeclaredMethods(),
									it -> it.getSimpleName().startsWith("create")
											&& IterableExtensions.isEmpty(it.getParameters())),
											&& IterableExtensions.isEmpty(it.getParameters())), MethodDeclaration::getSimpleName)
```
vs
```
FluentIterable.from(f.getDeclaredMethods()).filter(it-> it.getSimpleName().startsWith("create") && isEmpty(it.getParameters())).map(MethodDeclaration::getReturnType).toSortedList(Comparator.comparing(TypeReference::getSimpleName))
```